src/olympia/devhub/views.py

```diff
@@ -254,6 +254,15 @@ def _get_activities(request, action):
 
 
 def _get_items(action, addons):
+    if not isinstance(addons, (list, tuple)):
+        # MySQL 8.0.21 (and maybe higher) doesn't optimize the join with
+        # double # subquery the ActivityLog.objects.for_addons(addons) below
+        # would generate if addons is not transformed into a list first. Since
+        # some people have a lot of add-ons, we only take the last 100.
+        addons = list(
+            addons.all().order_by('-modified').values_list('pk', flat=True)[:100]
+        )
+
```

Comment:
I read this as the same scenario we were talking about on chat (that I thought you'd decided against):

>  So it wouldn't really be recent activity on the addons, it'd be activity on addons that have recently been updated.

expect that now rather being add-ons that have been updated in the past 6 months, it's the last 100 most recently modified add-ons (which for at least one prolific user is likely a lot less than 6 months).

If this is really a necessary restriction I think we need to call it out somewhere in the copy on the page.

Reply:
For 99% users there will be no change in behavior since they have less than 100 add-ons. For the most prolific ones I doubt the recent changes list was useful to them anyway (too many add-ons to care) so I don't think they'll notice, and I don't think we should draw attention to it on the page.